x-pack/plugins/observability_solution/observability_ai_assistant/public/components/insight/insight.tsx

```diff
@@ -192,20 +193,75 @@ function PromptEdit({
 }
 
 export interface InsightProps {
-  messages: Message[];
+  messages: Message[] | (() => Promise<Message[] | undefined>);
   title: string;
   dataTestSubj?: string;
 }
 
-export function Insight({ messages, title, dataTestSubj }: InsightProps) {
-  const [initialMessages, setInitialMessages] = useState(messages);
+enum FETCH_STATUS {
+  LOADING = 'loading',
+  SUCCESS = 'success',
+  FAILURE = 'failure',
+  NOT_INITIATED = 'not_initiated',
+}
+
+export function Insight({
+  messages: initialMessagesOrCallback,
+  title,
+  dataTestSubj,
+}: InsightProps) {
+  const [messages, setMessages] = useState<{ messages: Message[]; status: FETCH_STATUS }>({
+    messages: [],
+    status: FETCH_STATUS.NOT_INITIATED,
+  });
   const [isEditingPrompt, setEditingPrompt] = useState(false);
   const [isInsightOpen, setInsightOpen] = useState(false);
   const [hasOpened, setHasOpened] = useState(false);
   const [isPromptUpdated, setIsPromptUpdated] = useState(false);
 
-  const connectors = useGenAIConnectors();
+  const updateInitialMessages = useCallback(async () => {
```

Comment:
we also have useAsync or useAbortableAsync for this purpose IIRC

Reply:
`useAsync` resolves the function immediately. We want to resolve it only when the flyout opens.

Reply:
you can choose to not do anything though in the function if isOpen = false. But I'm not sure if that will be fugly. The only reason I would consider it because then we don't have to introduce FETCH_STATUS (we can use AsyncStatus)